### ISSUE TITLE
Docs reference outdated API

### ISSUE THREAD

**Issue by AdrianSchneble** [opentracing/opentracing-java#362]
The ["Scopes" doc](https://opentracing.io/guides/java/scopes/) references `Tracer.SpanBuilder.startActiveSpan(boolean finishOnClose)`, which [does not match the latest version](https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/Tracer.java) of OpenTracing.

As far as I can tell, the default is now simply that every span is set as active when `.start`ed. Regardless, the docs should be changed to reflect the latest version.

This is not the only issue with the docs, there are also numerous other examples of outdated example code (in the "Scopes" doc or other sections of the docs).

**Comment by whiskeysierra** [comment]
> As far as I can tell, the default is now simply that every span is set as active when `.start`ed.

That is not correct. The pattern is that span lifecycle (start/finish) is now properly separated from scope lifecycle (activate/close).

**Comment by MxKaras** [comment]
Given the changes, I'm trying to think under what conditions one would want to manage a scope's lifecycle. Is there any need to pay attention to it now?

**Comment by whiskeysierra** [comment]
Scopes are optional and users should be very interested in controlling the timeframe in which a span is active. It's not something that anyone should ignore.
